src/transxchange/read.rs

```diff
@@ -277,15 +283,139 @@ fn create_route(_transxchange: &Element, _vehicle_journey: &Element) -> Result<R
     })
 }
 
+struct NaiveDateRange {
+    current: NaiveDate,
+    end: NaiveDate,
+}
+
+impl NaiveDateRange {
+    fn new(start: NaiveDate, end: NaiveDate) -> NaiveDateRange {
+        if start <= end {
```

Comment:
Limit the endDate to Y+2 to avoid wtf dates (like 2099, 2500, 3019, 9999, etc.)

Reply:
Done in e46aba8bff9fd53a0615f10e39f42d16bc180ad5.